exporting/send_data.c

```diff
@@ -132,52 +153,47 @@ void simple_connector_receive_response(int *sock, struct instance *instance)
  * @param failures the number of communication failures.
  * @param instance an instance data structure.
  */
-void simple_connector_send_buffer(int *sock, int *failures, struct instance *instance)
+void simple_connector_send_buffer(
+    int *sock, int *failures, struct instance *instance, BUFFER *header, BUFFER *buffer, size_t buffered_metrics)
 {
-    BUFFER *buffer = (BUFFER *)instance->buffer;
-    size_t len = buffer_strlen(buffer);
-
     int flags = 0;
 #ifdef MSG_NOSIGNAL
     flags += MSG_NOSIGNAL;
 #endif
 
 #ifdef ENABLE_HTTPS
     uint32_t options = (uint32_t)instance->config.options;
-    struct opentsdb_specific_data *connector_specific_data = instance->connector_specific_data;
+    struct simple_connector_data *connector_specific_data = instance->connector_specific_data;
 
     if (options & EXPORTING_OPTION_USE_TLS)
         ERR_clear_error();
 #endif
 
     struct stats *stats = &instance->stats;
-
-    int ret = 0;
-    if (instance->send_header)
-        ret = instance->send_header(sock, instance);
-
     ssize_t written = -1;
+    size_t header_len = buffer_strlen(header);
+    size_t buffer_len = buffer_strlen(buffer);
 
-    if (!ret) {
 #ifdef ENABLE_HTTPS
-        if (instance->config.type == EXPORTING_CONNECTOR_TYPE_OPENTSDB_USING_HTTP &&
-            options & EXPORTING_OPTION_USE_TLS &&
-            connector_specific_data->conn &&
-            connector_specific_data->flags == NETDATA_SSL_HANDSHAKE_COMPLETE) {
-            written = (ssize_t)SSL_write(connector_specific_data->conn, buffer_tostring(buffer), len);
-        } else {
-            written = send(*sock, buffer_tostring(buffer), len, flags);
-        }
+    if (exporting_tls_is_enabled(instance->config.type, options) &&
+        connector_specific_data->conn &&
+        connector_specific_data->flags == NETDATA_SSL_HANDSHAKE_COMPLETE) {
+        written = (ssize_t)SSL_write(connector_specific_data->conn, buffer_tostring(header), header_len);
+        written += (ssize_t)SSL_write(connector_specific_data->conn, buffer_tostring(buffer), buffer_len);
+    } else {
+        written = send(*sock, buffer_tostring(header), header_len, flags);
+        written += send(*sock, buffer_tostring(buffer), buffer_len, flags);
```

Comment:
You should not try to send the `buffer`, if there was problem to send the `header`, please, add a test for the first test.

Reply:
Fixed.